aio/develop/run-development-container.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is a script that runs gulp in a docker container,
+# for machines that don't have nodejs, go and java installed.
+
+DOCKER_RUN_OPTS=${DOCKER_RUN_OPTS:-}
+CD="$(pwd)"
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+DASHBOARD_IMAGE_NAME=${K8S_DASHBOARD_CONTAINER_NAME:-"k8s-dashboard-dev-image"}
+K8S_DASHBOARD_SRC=${K8S_DASHBOARD_SRC:-"${CD}"}
+K8S_DASHBOARD_CONTAINER_NAME=${K8S_DASHBOARD_CONTAINER_NAME:-"k8s-dashboard-dev"}
+K8S_DASHBOARD_SRC_ON_CONTAINER=/go/src/github.com/kubernetes/dashboard
+
+echo "Remove existing container ${K8S_DASHBOARD_CONTAINER_NAME}"
+docker rm -f ${K8S_DASHBOARD_CONTAINER_NAME}
+
+# Always test if the image is up-to-date. If nothing has changed since last build,
+# it'll just use the already-built image
+echo "Start building container image for development"
+docker build -t ${DASHBOARD_IMAGE_NAME} -f ${DIR}/Dockerfile ${DIR}/../../
+
+# Run dashboard container for development and expose necessary ports automatically.
+# To keep container state as "Up", run `sh` as here
+echo "Start container for development"
+docker run \
+	-itd \
+	--net=host \
+	--name=${K8S_DASHBOARD_CONTAINER_NAME} \
+	-v /var/run/docker.sock:/var/run/docker.sock \
+	-v ${K8S_DASHBOARD_SRC}:${K8S_DASHBOARD_SRC_ON_CONTAINER} \
+	${DOCKER_RUN_OPTS} \
+	${DASHBOARD_IMAGE_NAME} \
+	sh
+
+# Cleanup backend source for build
+docker exec ${K8S_DASHBOARD_CONTAINER_NAME} rm -fr .tmp/*
+
+# Copy backend source for build
+docker exec ${K8S_DASHBOARD_CONTAINER_NAME} mkdir -p .tmp/src/github.com/kubernetes/dashboard/src/app
+docker exec ${K8S_DASHBOARD_CONTAINER_NAME} cp -r src/app/backend/ .tmp/src/github.com/kubernetes/dashboard/src/app
+docker exec ${K8S_DASHBOARD_CONTAINER_NAME} ln -s ${K8S_DASHBOARD_SRC_ON_CONTAINER}/vendor ${K8S_DASHBOARD_SRC_ON_CONTAINER}/.tmp/src/github.com/kubernetes/dashboard/src/app/backend/vendor
+
+# Copy kubeconfig into /root/.kube/config on the container
+docker exec ${K8S_DASHBOARD_CONTAINER_NAME} mkdir -p /root/.kube
+docker cp ${K8S_DASHBOARD_KUBECONFIG} ${K8S_DASHBOARD_CONTAINER_NAME}:/root/.kube/config
```

Comment:
It would be better to mount kubeconfig dir into the container. In case it changes we don't need to restart the container.

Reply:
Nice catch! I moved this into "docker run" command as -v option.